watertap/unit_models/leaky_pressure_exchanger.py

```diff
@@ -0,0 +1,491 @@
+                    b.low_pressure_side.properties_in[t].flow_vol)
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="mass leak from HP inlet to LP outlet")
+        def eq_mass_transfer_from_high_to_low_pressure_outlet(b, t, p, j):
+            comp = self.config.property_package.get_component(j)
+            if comp.is_solvent():
+                return (b.low_pressure_side.properties_out[t].flow_mass_phase_comp[p,j]==\
+                b.high_pressure_side.properties_in[t].flow_mass_phase_comp[p,j]*b.solvent_leakage_fraction[t]+
+                b.low_pressure_side.properties_in[t].flow_mass_phase_comp[p,j])
+            if comp.is_solute():
+                return (b.low_pressure_side.properties_out[t].flow_mass_phase_comp[p,j]==\
+                b.high_pressure_side.properties_in[t].flow_mass_phase_comp[p,j]*b.solute_leakage_fraction[t]+
+                b.low_pressure_side.properties_in[t].flow_mass_phase_comp[p,j])
+
+        ### this constraint (eq_mass_balance_high_pressure_outlet) might not be needed###
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="mass loss from HP inlet propagation to HP outlet")
+        def eq_mass_balance_high_pressure_outlet(b, t, p, j):
+            comp = self.config.property_package.get_component(j)
+            if comp.is_solvent():
+                return (b.high_pressure_side.properties_out[t].flow_mass_phase_comp[p,j]==\
+                b.high_pressure_side.properties_in[t].flow_mass_phase_comp[p,j]*(1-b.solvent_leakage_fraction[t]))
+            if comp.is_solute():
+                return (b.high_pressure_side.properties_out[t].flow_mass_phase_comp[p,j]==\
+                b.high_pressure_side.properties_in[t].flow_mass_phase_comp[p,j]*(1-b.solute_leakage_fraction[t]))
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Equal low pressure on both sides")
+        def eq_equal_low_pressure(b, t):
+            return (b.high_pressure_side.properties_out[t].pressure ==
+                    b.low_pressure_side.properties_in[t].pressure)
+
+        @self.low_pressure_side.Constraint(
+            self.flowsheet().config.time,
+            doc="Isothermal constraint")
+        def eq_isothermal_temperature(b, t):
+            return b.properties_in[t].temperature == b.properties_out[t].temperature
+
+        @self.high_pressure_side.Constraint(
+            self.flowsheet().config.time,
+            doc="Isothermal constraint")
+        def eq_isothermal_temperature(b, t):
+            return b.properties_in[t].temperature == b.properties_out[t].temperature
+
+    def initialize(
+            self,
+            state_args=None,
```

Comment:
We probably need to have two sets of state args - one each for the high and low pressure sides - as conditions in each side will be significantly different,